components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -8048,21 +8258,29 @@ public String getUUIDFromIdentifier(String provider, String apiName, String vers
     }
 
     /**
-     * Get API Product UUID by the API Product Identifier.
+     * Get API Product UUID by the API Product Identifier and organization.
      *
      * @param identifier API Product Identifier
+     * @param organization
      * @return String UUID
      * @throws APIManagementException if an error occurs
      */
-    public String getUUIDFromIdentifier(APIProductIdentifier identifier) throws APIManagementException {
-
+    public String getUUIDFromIdentifier(APIProductIdentifier identifier, String organization, Connection connection)
+            throws APIManagementException {
+        boolean isNewConnection = false;
         String uuid = null;
-        String sql = SQLConstants.GET_UUID_BY_IDENTIFIER_SQL;
-        try (Connection connection = APIMgtDBUtil.getConnection()) {
-            PreparedStatement prepStmt = connection.prepareStatement(sql);
+        PreparedStatement prepStmt = null;
+        String sql = SQLConstants.GET_UUID_BY_IDENTIFIER_AND_ORGANIZATION_SQL;
+        try {
+            if (connection == null) {
+                connection = APIMgtDBUtil.getConnection();
+                isNewConnection = true;
+            }
+            prepStmt = connection.prepareStatement(sql);
```

Comment:
Let's never follow this approach. We may not expect developers to read the implementation of the function to identify they can pass null into 'connection'. 
Also `connection` is a very sensitive parameter so we must make sure it is close properly. Having conditional things like `isNewConnection` can prone for bugs.

Lets use a new overloaded function which doesn't have `Connection connection` in the signature and make only that new method 'public'. This method which has `Connection connection` should only be called inside current class hence let's make it private.

Reply:
@tharikaGitHub

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10587